### PR TITLE
added tests for hashstate

### PR DESCRIPTION
**Pull request by Giulio2002**

### CODE REVIEW

**Review by yperbasis — CHANGES_REQUESTED** [review]
@Giulio2002 I've made a couple of fixes related to Issue #355, and now the test fails :( Could you investigate please.

**Comment by Giulio2002** [comment]
> @Giulio2002 I've made a couple of fixes related to Issue #355, and now the test fails :( Could you investigate please.

On my way🕵️

**Comment by Giulio2002** [comment]
Now it's fixed @yperbasis

File: node/silkworm/stagedsync/stage_hashstate_test.cpp [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+    db::stages::set_stage_progress(*txn, db::stages::kExecutionKey, 3);
+
+    CHECK(stagedsync::stage_hashstate(txn, data_dir.get_etl_path()) == stagedsync::StageResult::kSuccess);
+    CHECK(stagedsync::unwind_hashstate(txn, data_dir.get_etl_path(), 1) == stagedsync::StageResult::kSuccess);
+
+    auto hashed_address_table{db::open_cursor(*txn, db::table::kHashedAccounts)};
+
+    auto address_keccak{Bytes(keccak256(full_view(sender.bytes)).bytes, kHashLength)};
+
+    CHECK(hashed_address_table.seek(db::to_slice(address_keccak)));
```
Comment (yperbasis) [inline node/silkworm/stagedsync/stage_hashstate_test.cpp:240]: Please also check the value of the account.

**Review by yperbasis — APPROVED** [review]

**Pull request merged**